File: Solutions/Python/Easy/1539.py
class Solution:
    def findKthPositive(self, arr: List[int], k: int) -> int:
        n = len(arr)
        if arr[n - 1] - n < k:
            return k + n
        else:
            i, missing = 0, 0
            for j in range(1, arr[n - 1]):
                if j == arr[i]:
                    i += 1
                else:
                    missing += 1
                if missing == k:
                    return j

File: Solutions/C/Easy/1539.c
int findKthPositive(int* arr, int arrSize, int k){
}

/*
class Solution {
public:
    int findKthPositive(vector<int>& arr, int k) {
        int left = 0;
        int right = arr.size() - 1;
        while (left <= right) {
            int mid = left + (right - left) / 2;
            if (arr[mid] - (mid + 1) < k) {
                left = mid + 1;
            } else {
                right = mid - 1;
            }
        }
        return k + left;
    }
};
*/
